File: apps/web/src/types/testSuite.ts
export interface TestSuiteProject {
  id: number;
  project_code: string;
  name: string;
}

export interface TestSuiteTestCase {
  id: number;
  test_case_code: string;
  title: string;
}

export interface TestSuite {
  id: number;
  suite_code: string;
  project_id: number;
  project: TestSuiteProject;
  test_cases: TestSuiteTestCase[];
  name: string;
  description: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface TestSuiteRequest {
  project_id: number;
  name: string;
  description: string | null;
  status: string;
}

File: apps/api/app/schemas/test_suite.py
from datetime import datetime

from pydantic import BaseModel


class TestSuiteProjectResponse(BaseModel):
    id: int
    project_code: str
    name: str

    class Config:
        from_attributes = True


class TestSuiteBase(BaseModel):
    project_id: int
    name: str
    description: str | None = None
    status: str


class TestSuiteCreate(TestSuiteBase):
    pass


class TestSuiteUpdate(TestSuiteBase):
    pass


class TestSuiteTestCaseResponse(BaseModel):
    id: int
    test_case_code: str
    title: str

    class Config:
        from_attributes = True


class TestSuiteResponse(TestSuiteBase):
    id: int
    suite_code: str
    project: TestSuiteProjectResponse
    test_cases: list[TestSuiteTestCaseResponse] = []
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True